### PR TITLE
FIX/Refactor: continuous_tests to class

### PR DESCRIPTION
**Pull request by juaninf**

### CODE REVIEW

File: claasp/cipher_modules/continuous_tests.py [hunk 1]
```diff
@@ -30,439 +29,457 @@
...
-            if (component["type"] == "intermediate_output" or component["type"] == "cipher_output") and \
-                    component["description"] not in component_descriptions:
-                component_descriptions.append(component["description"])
-
-                temp_components[component["description"][0]] = list(range(component["output_bit_size"]))
-
-    return temp_components
-
-
-def continuous_avalanche_factor(cipher, lambda_value, number_of_samples):
```
Comment (peacker) [inline claasp/cipher_modules/continuous_tests.py:259]: add doctest documentation

File: claasp/cipher_modules/continuous_tests.py [hunk 2]
```diff
@@ -30,439 +29,457 @@
...
-        index_of_tag_input = cipher.inputs.index(tag_input)
-        lst_input = []
-        i = 0
-        for _ in cipher.inputs:
-            if i == index_of_tag_input:
-                lst_input.append([])
-            else:
-                lst_input.append([Decimal(max_bias[random.randrange(0, 2)]) for _ in range(cipher.inputs_bit_size[i])])
-            i += 1
+class ContinuousDiffusionAnalysis:
```
Comment (peacker) [inline claasp/cipher_modules/continuous_tests.py:32]: Call it ContinuousDiffusionTests

File: claasp/cipher.py [hunk 3]
```diff
@@ -146,6 +147,7 @@ def __init__(self, family_name, cipher_type, cipher_inputs,
         self._reference_code = cipher_reference_code
         self._id = self.make_cipher_id()
         self._file_name = self.make_file_name()
+        self._continuous_tests = ContinuousDiffusionAnalysis(self)
```
Comment (peacker) [inline claasp/cipher.py:150]: remove references to continuous diffusion from cipher.py

File: claasp/cipher_modules/continuous_tests.py [hunk 4]
```diff
@@ -30,439 +29,457 @@
...
+                input_size = self.cipher.inputs_bit_size[ii]
+                if ii == index_of_tag_input:
+                    lst_input.append([])
                 else:
-                    df_values[index][key] += continuous_neutrality_measures_output_values_[index][key]
-            else:
-                df_values[index][key] = 0.0
-
-
-def continuous_diffusion_tests(cipher,
```
Comment (peacker) [inline claasp/cipher_modules/continuous_tests.py:318]: add documentation from cipher.py

**Review by peacker — CHANGES_REQUESTED** [review]

**Comment by juaninf** [comment]
Declined in favor of Convert continuous diffusion tests to class #158

**Pull request closed without merging**